extensions/amp-sidebar/1.0/sidebar-hooks.js

```diff
@@ -0,0 +1,198 @@
+      ) {
+        // reverse the current animation
+        const sidebarAnimation = sidebarAnimationRef.current;
+        if (sidebarAnimation) {
+          sidebarAnimation.reverse();
+          sidebarAnimation.onfinish = opened
+            ? postVisibleAnim
+            : postInvisibleAnim;
+        }
+        backdropAnimationRef.current && backdropAnimationRef.current.reverse();
+        animationDirectionRef.current = opened
+          ? Direction.OPENING
+          : Direction.CLOSING;
+      }
+      return;
+    }
+
+    // currently in fully opened or closed state
+    if (opened) {
+      // make visible animation
+      if (!sidebarElement.animate || !backdropElement.animate) {
+        postVisibleAnim();
+        return;
+      }
+      safelySetStyles(
+        sidebarElement,
+        side === Side.LEFT
+          ? ANIMATION_STYLES_SIDEBAR_LEFT_INIT
+          : ANIMATION_STYLES_SIDEBAR_RIGHT_INIT
+      );
+      safelySetStyles(backdropElement, ANIMATION_STYLES_BACKDROP_INIT);
+      const sidebarAnimation = sidebarElement.animate(
+        side === Side.LEFT
+          ? ANIMATION_KEYFRAMES_SLIDE_IN_LEFT
+          : ANIMATION_KEYFRAMES_SLIDE_IN_RIGHT,
+        {
+          duration: ANIMATION_DURATION,
+          fill: 'both',
+          easing: ANIMATION_EASE_IN,
+        }
+      );
+      sidebarAnimation.onfinish = postVisibleAnim;
+      const backdropAnimation = backdropElement.animate(
+        ANIMATION_KEYFRAMES_FADE_IN,
+        {
+          duration: ANIMATION_DURATION,
+          fill: 'both',
+          easing: ANIMATION_EASE_IN,
+        }
+      );
+      sidebarAnimationRef.current = sidebarAnimation;
+      backdropAnimationRef.current = backdropAnimation;
+      animationDirectionRef.current = Direction.OPENING;
+    } else {
+      // make invisible animation
+      if (!sidebarElement.animate || !backdropElement.animate) {
+        postInvisibleAnim();
+        return;
+      }
+      const sidebarAnimation = sidebarElement.animate(
+        side === Side.LEFT
+          ? ANIMATION_KEYFRAMES_SLIDE_IN_LEFT
+          : ANIMATION_KEYFRAMES_SLIDE_IN_RIGHT,
+        {
+          duration: ANIMATION_DURATION,
+          direction: 'reverse',
+          fill: 'both',
+          easing: ANIMATION_EASE_IN,
+        }
+      );
+      sidebarAnimation.onfinish = postInvisibleAnim;
+      const backdropAnimation = backdropElement.animate(
+        ANIMATION_KEYFRAMES_FADE_IN,
+        {
+          duration: ANIMATION_DURATION,
+          direction: 'reverse',
+          fill: 'both',
+          easing: ANIMATION_EASE_IN,
+        }
+      );
+      sidebarAnimationRef.current = sidebarAnimation;
+      backdropAnimationRef.current = backdropAnimation;
+      animationDirectionRef.current = Direction.CLOSING;
+    }
+  }, [
+    opened,
+    onAfterCloseRef,
+    side,
+    sidebarRef,
+    backdropRef,
+    setMounted,
+    sidebarAnimationRef,
```

Comment:
The local refs should not be in the deps list.

Reply:
Thank you, removed!